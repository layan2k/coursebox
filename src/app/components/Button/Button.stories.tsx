import type { Meta, StoryFn } from '@storybook/react';
import { Button } from "./Button";
import React from 'react';

const meta = {
    title: "Controls/Button",
    component: Button,
    args: {
        children:'Click Me'
    }
} as Meta<typeof Button>

export default meta

export const Primary = {
    args : {
        primary: true,
    }
}

export const LargePrimary = {
    args: {
        ...Primary.args,
        size: 'large',
        children: "More To Click Me"
    }
}




