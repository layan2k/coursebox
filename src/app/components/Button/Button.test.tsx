import React from "react"
import { fireEvent, render, screen } from '@testing-library/react'
import { Button } from "./Button"

describe("Button test cases ", () => {
    it('Render check', () => {
        const onclick = jest.fn();
        const { asFragment } = render(<Button onClick={onclick}>Button</Button>);
        expect(asFragment()).toMatchSnapshot();
    })
    it("Check onClick callback", () => {
        const onclick = jest.fn();
        render(<Button onClick={onclick}>Button</Button>);
        const element = screen.getByRole("button");
        console.log("Button element:", element);
        fireEvent.click(element);
        expect(onclick).toHaveBeenCalled()
    })
})