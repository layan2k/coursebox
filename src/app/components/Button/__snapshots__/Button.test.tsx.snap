// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases  Render check 1`] = `
<DocumentFragment>
  <button
    class="css-pbyf8u"
    color="primary"
  >
    Button
  </button>
</DocumentFragment>
`;
